### DISCUSSION TITLE
Search match count

### DISCUSSION THREAD

**Discussion by evan-zeller-mesh** [Kong/insomnia#3642] [Ideas]
- Insomnia Version: 5.14.7
- Operating System: elementary OS 0.4.1 Loki (Ubuntu 16.04.2 LTS)

## Details

Insomnia has become my go to REST client aside from one feature, which is finding how many occurrences of a given value are in a response. It's a quick and dirty method of verification commonly used for the software I work with and for this purpose I usually keep open Insomnia and Postman. If a simple match count was added for ctrl-f searches I could completely eradicate Postman from my system. :smile:

**Comment by gschier** [comment]
Hmm, this seems like a fairly simple request. 

It doesn't look like Codemirror supports this feature but maybe we could add it ourselves: https://github.com/codemirror/CodeMirror/issues/3728

**Comment by Vyoam** [comment]
For note, Ace seems to support it. A bit weird how this request was outright closed in CodeMirror.
https://github.com/ajaxorg/ace/issues/3121
https://ace.c9.io

**Reply by svrooij** [reply]
It was not outright closed, it stated that it would be solved in some other PR.
>  https://github.com/ajaxorg/ace/issues/3121 should fix this. btw where is this screenshot from?

**Comment by alinoancea** [comment]
Hello @gschier,

Is this feature is still available/open for implementation? If so, I would like to give it a try.

Thanks.

**Comment by gschier** [comment]
Sorry for the delayed response but yes it is @alinoancea

**Comment by svrooij** [comment]
Any ideas on this issue? It seems like a really small change with so much to gain. 

How would I get started in implementing this?

**Reply by wdawson** [reply]
Hey @svrooij ! Let me see if there's a dev on the team that might be able to point you in the right direction. We would love a contribution for this!

**Reply by dimitropoulos** [reply]
Please check my work because I'm hoping there's a simple thing I'm missing here, but it doesn't appear like this is easy to do, unfortunately.  That search dialog is an out-of-the-box (and unconfigured, on our side) feature of CodeMirror.  We're not in codemirror 6 yet, but here are the docs for it in codemirror 6: https://codemirror.net/6/docs/ref/#search.  Looking at that, I don't see a way to turn on a count.  Looking at the source code (https://github.com/codemirror/CodeMirror/blob/master/addon/search/search.js) I don't see it hiding in there, either.

I turned to Google and the best result I found was https://stackoverflow.com/questions/58290588/how-can-i-add-count-for-the-search-results-into-the-codemirror which basically says "mutate the DOM manually and add it", which, although technically possible, is not the kind of thing I think we would do in insomnia.

You don't happen to be super knowledgeable on [monaco](https://microsoft.github.io/monaco-editor/), do you?  I'm hoping that in time we'll be able to switch to it, and this would be one of a great many issues that would be solved automagically if we were to do so.

**Comment by manisha-singh-hyd** [comment]
I made one. Please give it a try and let me know if it works for you. I would be happy to bring in changes if required. You can install it from insomnia pulgins
insomnia-plugin-response-search-counter

Github- https://github.com/manisha-singh-hyd/insomnia-plugin-response-search-counter

![search-counter](https://github.com/user-attachments/assets/02489e47-2b0f-41ab-b384-ffacc9dd3886)

**Reply by svrooij** [reply]
Looks nice, but I'm moved on to something integrated in my ide.
